tests/genai/test_git_versioning.py

```diff
@@ -0,0 +1,90 @@
+import subprocess
+from pathlib import Path
+from unittest import mock
+
+import pytest
+
+from mlflow.genai import disable_git_model_versioning, enable_git_model_versioning
+from mlflow.genai.git_versioning import _get_active_git_context
+from mlflow.genai.git_versioning.git_info import GitOperationError
+
+
+@pytest.fixture(autouse=True)
+def cleanup_active_context():
+    yield
+    disable_git_model_versioning()
+
+
+@pytest.fixture
+def tmp_git_repo(tmp_path: Path, monkeypatch: pytest.MonkeyPatch):
+    path = tmp_path / "test_repo"
+    path.mkdir()
+    subprocess.check_call(["git", "init"], cwd=path)
+    subprocess.check_call(["git", "config", "user.name", "test"], cwd=path)
+    subprocess.check_call(["git", "config", "user.email", "test@example.com"], cwd=path)
+    subprocess.check_call(["git", "commit", "--allow-empty", "-m", "test"], cwd=path)
+    monkeypatch.chdir(path)
+    return path
+
+
+def test_enable_git_model_versioning(monkeypatch: pytest.MonkeyPatch, tmp_git_repo: Path):
+    context = enable_git_model_versioning()
+    assert context.info.commit is not None
+    assert context.info.branch is not None
+    assert context.info.dirty is False
+
+    # Create a dummy file to make the repo dirty
+    Path(tmp_git_repo / "dummy.txt").touch()
+    context = enable_git_model_versioning()
+    assert context.info.dirty is True
+
```

Comment:
We shouldn't have to re-create a context in order to get `context.info.dirty is True` after creating a tracked file, right?

(Note that this should only return True if a tracked file has uncommitted changes - https://github.com/mlflow/mlflow/pull/16923/files#r2240273865)

Reply:
This file is still untracked. `dirty` should remain false, right?